BuildingBlocks/src/BuildingBlocks.Infrastructure/Persistence/BlobStorage/GoogleCloudStorage/GoogleCloudStorage.cs

```diff
@@ -102,13 +107,17 @@ private async Task UploadChangedBlobs()
         {
             await EnsureKeyDoesNotExist(blob.Folder, blob.Name);
 
-            await using var memoryStream = new MemoryStream(blob.Content);
+            var memoryStream = await _logger.TraceTime(async () => new MemoryStream(blob.Content), nameof(MemoryStream));
 
             try
             {
                 _logger.LogTrace("Uploading blob with key '{blobName}'...", blob.Name);
-                await _storageClient.UploadObjectAsync(blob.Folder, blob.Name, null,
-                    memoryStream);
+
+                await _logger.TraceTime(async () =>
+                {
+                    await _storageClient.UploadObjectAsync(blob.Folder, blob.Name, null, memoryStream);
```

Comment:
Can you also log the time for the whole `UploadChangedBlobs` method please? (in `SaveAsync`, wrap the call of the method)

Reply:
Time logged for the whole `UploadChangedBlobs()` method.